assets/js/util/test/extractForSparkline.js

```diff
@@ -114,8 +114,128 @@ const valuesToTest = [
 	],
 ];
 
+const columnIndexValuesToTest = [
+	[
+		[
+			[
+				'1/1/2019',
+				1,
+				2,
+				3,
+			],
+			[
+				'1/2/2019',
+				4,
+				5,
+				6,
+			],
+		],
+		0,
+		[
+			[ '1/1/2019', '1/1/2019' ], [ '1/2/2019', '1/2/2019' ],
+		],
+	],
+	[
+		[
+			[
+				'1/1/2019',
+				1,
+				2,
+				3,
+			],
+			[
+				'1/2/2019',
+				4,
+				5,
+				6,
+			],
+		],
+		1,
+		[
+			[ 1, '1/1/2019' ], [ 4, '1/2/2019' ],
+		],
+	],
+	[
+		[
+			[
+				'1/1/2019',
+				1,
+				2,
+				3,
+			],
+			[
+				'1/2/2019',
+				4,
+				5,
+				6,
+			],
+		],
+		2,
+		[
+			[ 2, '1/1/2019' ], [ 5, '1/2/2019' ],
+		],
+	],
+	[
+		[
+			[
+				'1/1/2019',
+				1,
+				2,
+				3,
+			],
+			[
+				'1/2/2019',
+				4,
+				5,
+				6,
+			],
+		],
+		3,
+		[
+			[ 3, '1/1/2019' ], [ 6, '1/2/2019' ],
+		],
+	],
+];
+
 describe( 'extractForSparkline', () => {
 	it.each( valuesToTest )( 'for start date %s and end date %s should returns %s', ( data, column, expected ) => {
 		expect( extractForSparkline( data, column ) ).toStrictEqual( expected );
 	} );
+
+	it.each( columnIndexValuesToTest )( 'for start date %s and columnIndex %s it should return %s', ( data, columnIndex, expected ) => {
+		expect( extractForSparkline( data, 0, columnIndex ) ).toStrictEqual( expected );
+	} );
+	const data = [
+		{
+			dimensions: [
+				'Site Kit Test 1',
+				'/',
+			],
+			metrics: [
+				{
+					values: [
+						'1469',
+					],
+				},
+			],
+		},
+		{
+			dimensions: [
+				'Site Kit Test 2',
+				'/site-kit-tests/',
+			],
+			metrics: [
+				{
+					values: [
+						'616',
+					],
+				},
+			],
+		},
+	];
+
+	it( 'returns data from an index passed as a string', () => {
+		expect( extractForSparkline( data, 0, 'metrics.0.values.0' ) ).toStrictEqual( [ [ '1469', '' ], [ '616', 0 ] ] );
+		expect( extractForSparkline( data, 1, 'dimensions.1' ) ).toStrictEqual( [ [ '/', '' ], [ '/site-kit-tests/', 0 ] ] );
```

Comment:
The `0` and `1` here as the second argument to `extractForSparkline` is the path to the y-axis value. In this case, neither exists which results in an odd looking result.

The y value from the first pair is an empty string because the first row is treated differently but we can ignore that for now. The y value from the second pair is `0` for both again because neither row has a property of `0` so the fallback value is `0` for both.

Let's update the data used to have a more generic structure and update the test show the relationship between the data and the path arguments more clearly.

e.g.
```js
const data = [
	{ key: 'foo', value: 123 },
	{ key: 'bar', value: 456 },
]
extractForSparkline( data, 'value', 'key' ) // [ [ 'foo', 123 ], [ 'bar', 456 ] ]
```
Another test to show that object paths `a.b.c.` are supported for both arguments as well would be great